mne/io/edf/tests/test_one_channel_gdf.py

```diff
@@ -0,0 +1,361 @@
+# Author: Abram Hindle <hindle1@ualberta.ca>
+# Adapted from:
+#   Author: Eric Larson <larson.eric.d@gmail.com>
+#           Adapted from vispy
+#
+# License: BSD (3-clause)
+
+from os import path as op
+
+import pytest
+import mne
+import base64
+
+from mne.utils import run_tests_if_main, _TempDir
+
+# the file in question doesn't have highpass and lowpass
+@pytest.mark.filterwarnings('ignore::RuntimeWarning')
+def test_one_channel_gdf():
+    """Test line endings of mne-python."""
+    tempdir = _TempDir()
+    fname = op.join(tempdir, 'single_channel.gdf')
+    with open(fname, 'wb') as fid:
+        fid.write(one_channel_gdf())
+    # pytest.raises(AssertionError, _assert_line_endings, tempdir)
+    # now check mne
+    ecg = mne.io.read_raw_gdf(fname, preload=True)
+    assert ecg['ECG'][0].shape == (1, 4500)
+    assert 150.0 == ecg.info['sfreq']
+
+
+def one_channel_gdf():
+    ''' This is an embedded 1 channel ECG '''
+    b64data = """R0RGIDIuMTAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
```

Comment:
It would probably be better to have this in a separate file, esp. in the `mne-testing-data` repo (see for instructions)

https://github.com/mne-tools/mne-testing-data